hack/generated/pkg/reflecthelpers/reflect_helpers.go

```diff
@@ -208,3 +181,75 @@ func FindResourceReferences(transformer genruntime.ARMTransformer) (map[genrunti
 
 	return result, nil
 }
+
+type resolved struct {
+	ResolvedDetails   genruntime.ConvertToARMResolvedDetails
+	ResourceHierarchy genruntime.ResourceHierarchy
+}
+
+func resolve(
```

Comment:
This boundary definitely seems like a good one for a function since you don't need to pass too many things in. I'm trying to think of better names for things though. It seems like the `resolved` struct isn't buying you anything, and getting rid of it would free us from needing to come up with a better name for it! 

What about putting the resource hierarchy into `genruntime.ConvertToARMResolvedDetails` and just returning that? It doesn't seem like the struct is used for anything else where the hierarchy wouldn't be available, and it makes sense to have it in there.

If you're not keen on that then maybe just return details, hierarchy and error separately.

Reply:
Now that you've extracted this it seems like it should be a method on `genruntime.Resolver` - it ends up being the entrypoint that drives the rest of the resolver, turning a `MetaObject` into a `ConvertToARMResolvedDetails`.

Reply:
Still not sure on the name though - `Resolver.ResolveObject` seems a bit... plodding. `Resolver.FullyResolve` is ok though? I guess part of the problem is that the type is named for the thing it does, and we want it to do that thing.

Reply:
`Resolver.DoItTo(object)`

Reply:
I came up with `Resolver.ResolveForARM`, which isn't terrible? I do agree that this method feels like it should be on `Resolver`. I end up running into a bit of a package dependency problem though:
1. `Resolver.ResolveForARM` uses the `FindResourceReferences` function, which in turn uses `ReflectVisitor`, both of which are defined in the `reflecthelpers` package.
2. `reflecthelpers` has a dependency on `genruntime`, partially for its use of `Resolver` and partially for `MetaObject`.

So there's a circular dependency here, possibly being caused by unclear package purpose (`genruntime` is so much of a catch-all that everything in `reflecthelpers` could in theory go into it too...)

I could move `reflect_helpers.go` into `genruntime`, and just leave `reflect_visitor` in `reflecthelpers`, which would solve this problem... and possibly also be closer to the admitted "reality" right now which is that `genruntime` and `reflecthelpers` are pretty tightly related